src/lib/components/Cytoscape.react.js

```diff
@@ -798,7 +810,84 @@ Cytoscape.propTypes = {
+     *     - `commands` (dictionary): An array of commands to list in the menu or a function that returns the array.
+     *         - `id` (string): Id that is returned in the ctxmenu callback to identify specific command triggered. Required.
+     *         - `content` (string): HTML/text content to be displayed in the menu. Required.
+     *         - `contentStyle` (dictionary): CSS key:value pairs to set the command's css in js if you want.
+     *         - `enabled` (bool): Whether the command is selectable.
+     *         - `format` (array): Data structure which determines how raw information from cytoscape's json function is filtered
+     *     - `menuRadius` (number): The radius of the circular menu in pixels.
+     *     - `fillColor` (string): The background colour of the menu.
+     *     - `activeFillColor` (string): The colour used to indicate the selected command.
+     *     - `activePadding` (number): Additional size in pixels for the active command.
+     *     - `indicatorSize` (number): The size in pixels of the pointer to the active command.
+     *     - `separatorWidth` (number): Elements matching this Cytoscape.js selector will trigger cxtmenus.
+     *     - `spotlightPadding` (number): Extra spacing in pixels between the element and the spotlight.
+     *     - `minSpotlightRadius` (number): The minimum radius in pixels of the spotlight.
+     *     - `maxSpotlightRadius` (number): The maximum radius in pixels of the spotlight.
+     *     - `openMenuEvents` (string): Space-separated cytoscape events that will open the menu; only `cxttapstart` and/or `taphold` work here.
+     *     - `itemColor` (string): The colour of text in the command's content.
+     *     - `itemTextShadowColor` (string): The text shadow colour of the command's content.
+     *     - `zIndex` (number): The z-index of the ui div.
+     *     - `atMouse` (bool): Draw menu at mouse position.
+     *
+     * 2. The format array inside the `commands` dictionary which determines how information is passed back.
+     * In the format array, strings can be entered to determine which properties of the cytoscape json object are selected.
+     * Alternatively, instead of a string, a dictionary can be used to select properties more specifically. The properties of such an object are outlined below:
+     *     - `key` (string): Specifies the property to be selected in the json object.
+     *     - `props` (array): This array specifies the properties of the key which are selected.
+     *     - `filter` (dictionary): This property is only to be applied where the value of the `key` is an array of objects. This object specifies the properties and values that the object in the array must have to not be removed from the array.
+     * For an example of the capabilities of this system, please reference the ctxmenu demo.
+     */
+    ctxmenu: PropTypes.arrayOf(
+        PropTypes.shape({
```

Comment:
Unless you want to accept arbitrary additional keys, use `PropTypes.exact` instead of `PropTypes.shape`.

And rather than documenting the whole object up above, you can document each piece as it comes. See for example [dash-table columns prop](https://github.com/plotly/dash-table/blob/0aeab39a001d530645ce1f017ef6cfd6f1f683f9/src/dash-table/dash/DataTable.js#L130)